Sync/State.swift

```diff
@@ -473,6 +463,23 @@ open class Scratchpad {
 
         b.hashedUID = prefs.stringForKey(PrefHashedUID)
 
+        b.fxaDeviceId = prefs.stringForKey(PrefDeviceID) ?? {
+            // Migrate from previous way of storing device id.
+            // This code will only be run once – the id will be stored
+            // in PrefDeviceID.
+            let PrefDeviceRegistration = "deviceRegistration"
+            if let string = prefs.stringForKey(PrefDeviceRegistration) {
+                let json = JSON(parseJSON: string)
+                if let id = json["id"].string {
+                    return id
```

Comment:
If we subsequently unregister, I think we clear the whole prefs branch, but would it be wise to remove `"deviceRegistration"` from prefs before returning?

Reply:
Removed the value for the `"deviceRegistration"` key.